tests/test_barker.py

```diff
@@ -0,0 +1,171 @@
+
+    # system with a large number of quartic oscillators
+    x = np.zeros(1_000)
+
+    prop = BarkerProposal(grad_log_q, proposal_sig=0.1)
+
+    def mcmc_move(x):
+        y = prop.sample(x)
+
+        log_prob_fwd = prop.log_density(x, y)
+        log_prob_rev = prop.log_density(y, x)
+
+        _log_accept_prob = log_q(y) - log_q(x) + log_prob_rev - log_prob_fwd
+        accept_prob = np.exp(min(0.0, _log_accept_prob))
+
+        if np.random.rand() < accept_prob:
+            return y
+        else:
+            return x
+
+    _traj = [x]
+    for _ in range(2_000):
+        _traj.append(mcmc_move(_traj[-1]))
+
+    samples = np.array(_traj[100:]).flatten()
+
+    # summarize using histogram
+    y_empirical, edges = np.histogram(samples, bins=100, range=(-2, +2), density=True)
+    x_grid = (edges[1:] + edges[:-1]) / 2
+
+    # compare with ref
+    y = np.exp(np.array([log_q(x) for x in x_grid]))
+    y_ref = y / np.trapz(y, x_grid)
+
+    histogram_mse = np.mean((y_ref - y_empirical) ** 2)
+
+    assert histogram_mse < threshold
+
+
+@pytest.mark.nogpu
+@pytest.mark.parametrize("proposal_sig", [0.1, 1.0])
+def test_proposal_magnitude_independent_of_force_magnitude(proposal_sig):
+    """Generate Lennard-Jones-informed proposals from clashy vs. relaxed starting points
+        (where |force(x_clash)| ~= +inf and |force(x_relaxed)| ~ 0).
+
+    Assert that:
+    * the avg. squared distance between proposal and starting point is the same in both cases
+    * the proposal skew is ~ 100% in the |force| ~= +inf case, and ~ 0% in the |force| ~= 0 case
+    """
+
+    def log_q(r):
+        sig, eps = 1.0, 1.0
+        return jnp.sum(-4 * eps * ((sig / r) ** 12 - (sig / r) ** 6))
+
+    grad_log_q = jit(grad(log_q))
+
+    barker_prop = BarkerProposal(grad_log_q, proposal_sig=proposal_sig)
+    expected_sq_distance = barker_prop.proposal_sig ** 2
+
+    # ---------------------
+    np.random.seed(0)
+
+    # sample many proposals from a clashy initial condition
+    x_clash = np.array([1e-3])
+    assert np.linalg.norm(grad_log_q(x_clash)) > 1e10
+    ys_clash = np.array([barker_prop.sample(x_clash) for _ in range(10_000)])
+
+    # assert that the gradient-informed proposals
+    # are the same avg. sq. distance from starting point
+    # as if proposed from the base kernel Normal(mu=x_clash, sig=proposal_sig)
+    disp_clash = (ys_clash - x_clash).flatten()
+    mean_sq_distance_clash = (disp_clash ** 2).mean()
+
+    assert pytest.approx(mean_sq_distance_clash, rel=1e-1) == expected_sq_distance
+
+    # assert that the proposals are skewed in the expected direction
+    skew = np.sign(disp_clash).mean()
+    assert pytest.approx(skew, abs=1e-2) == 1.0
+
+    # ---------------------
+    np.random.seed(0)
+
+    # sample many proposals from a relaxed initial condition
+    x_relaxed = np.array([1e3])
+    assert np.linalg.norm(grad_log_q(x_relaxed)) < 1e-10
+    ys = np.array([barker_prop.sample(x_relaxed) for _ in range(10_000)])
```

Comment:
nit: this is ~1000x faster in my testing (probably not important, but does reduce test duration from 15s to ~0.1s)
```suggestion
    ys = barker_prop.sample(np.tile(x_relaxed, 10_000))
```

Reply:
Ahh, good catch -- in this case these two should be identical (1 proposal on 10000D state vs. 10000 independent proposals on 1D state...)

Reply:
>Hmm, is it ever the case that they would be different?

Depends how we define `log_q` / `grad_log_q`.

(In the test, defined in such a way that `grad_log_q(xs) == [grad_log_q(x) for x in xs]`, but this may not be the case for other user-defined functions `grad_log_q`.)

Reply:
Applied in https://github.com/proteneer/timemachine/pull/978/commits/96e035df2074527faf9013dfd99284ddd38fba45